proto/site/v1alpha/site.proto

```diff
@@ -0,0 +1,207 @@
+syntax = "proto3";
+
+package com.mintter.site.v1alpha;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+option go_package = "mintter/backend/genproto/site/v1alpha;site";
+
+// Site API service allows to configure a remote Mintter Site.
+// The server exposing this API should take care about authenticating the caller.
+service Site {
+  // Creates a new invite token for registering a new member.
+  rpc CreateInviteToken(CreateInviteTokenRequest) returns (InviteToken);
+
+  // Redeem a previously created invite token to register a new member.
+  rpc RedeemInviteToken(RedeemInviteTokenRequest) returns (RedeemInviteTokenResponse);
+
+  // Gets public-facing site information.
+  rpc GetSiteInfo(GetSiteInfoRequest) returns (SiteInfo);
+
+  // Updates public-facing site information. Doesn't support partial updates,
+  // hence all the fields must be provided.
+  rpc UpdateSiteInfo(UpdateSiteInfoRequest) returns (SiteInfo);
+
+  // Lists registered members on the site.
+  rpc ListMembers(ListMembersRequest) returns (ListMembersResponse);
+
+  // Gets information about a specific member.
+  rpc GetMember(GetMemberRequest) returns (Member);
+
+  // Deletes an existing member.
+  rpc DeleteMember(DeleteMemberRequest) returns (google.protobuf.Empty);
+
+  // Blocks a given Mintter Account.
+  rpc BlockAccount(BlockAccountRequest) returns (BlockAccountResponse);
+
+  // Unblock a previously blocked Mintter Account.
+  rpc UnblockAccount(UnblockAccountRequest) returns (UnblockAccountResponse);
```

Comment:
Does this means that the editors you add to the site will be there forever? because Block can be the same as removing if from the list and Unblock as adding it to the editor's list right?

Reply:
There's a DeleteMember method to delete. Maybe block is not even necessary. Threw that in just in case. Though that maybe we want to block someone from even ever redeeming the invite, or something like that. But maybe delete is enough.

WDYT @ericvicenti @juligasa?

Reply:
I think `DeleteMember` is enough. If you want to reinstate a previously deleted member, just create a new invitation link and send it again... As for the unused invitation links, as they may eventually expire, unredeemed links are not a threat either

Reply:
Maybe I was thinking about blocking for cases where you'd delete a member, and then want to make sure it never gets back accidentally. Although since inviting is an explicit action, maybe there's no need for any of this blocking stuff. Vote on this comment whether it should be removed :)

Reply:
Removed